packages/relay/src/lib/eth.ts

```diff
@@ -504,11 +504,50 @@ export class EthImpl implements Eth {
 
     try {
       let weibars: BigNumber | number = 0;
-
       const mirrorAccount = await this.mirrorNodeClient.getAccount(account, requestId);
-      if (mirrorAccount && mirrorAccount.balance) {
-        weibars = mirrorAccount.balance.balance * constants.TINYBAR_TO_WEIBAR_COEF
-      } else {
+      let balanceFound = false;
+
+      // A blockNumberOrTag has been provided. If it is `latest` or `pending` retrieve the balance from /accounts/{account.id}
+      if (mirrorAccount && blockNumberOrTag !== null && blockNumberOrTag !== EthImpl.blockLatest && blockNumberOrTag !== EthImpl.blockPending) {
```

Comment:
Move the blockNumberTag checks to a method.
I think the same check for a valid non latest value exists somewhere else also